src/Microsoft.Dnx.Runtime/DependencyManagement/LibraryManager.cs

```diff
@@ -78,8 +101,9 @@ private void EnsureInitialized()
                 if (!_initialized)
                 {
                     _initialized = true;
-                    _graph = _libraryInfoThunk().ToDictionary(ld => ld.Name,
-                                                              StringComparer.Ordinal);
+                    var libraries = _librariesThunk();
+                    _graph = libraries.ToDictionary(l => l.Identity.Name, StringComparer.Ordinal);
```

Comment:
Actually, I think we can combine these into a single dictionary storing a tuple (or a custom class). We need to be able to retrieve both the externally-used `Library` class and the internal `LibraryDescription`. I can merge the two into one dictionary though, either by putting `Library` on `LibraryDescription` or creating a tuple.